tests/python/pytorch/graphbolt/test_feature_fetcher.py

```diff
@@ -0,0 +1,41 @@
+import dgl
+import dgl.graphbolt
+import pytest
+import torch
+
+
+def get_graphbolt_fetch_func():
+    feature_store = dgl.graphbolt.feature_store.InMemoryFeatureStore(
+        {
+            "feature": torch.randn(200, 4),
+            "label": torch.randint(0, 10, (200,)),
+        }
+    )
+
+    def fetch_func(data):
+        return feature_store.read("feature", data), feature_store.read(
+            "label", data
+        )
+
+    return fetch_func
+
+
+def get_tensor_fetch_func():
+    feature_store = torch.randn(200, 4)
+    label = torch.randint(0, 10, (200,))
+
+    def fetch_func(data):
+        return feature_store[data], label[data]
+
+    return fetch_func
+
+
+@pytest.mark.parametrize(
+    "fetch_func", [get_graphbolt_fetch_func(), get_tensor_fetch_func()]
+)
+def test_FeatureFetcher(fetch_func):
+    itemset = dgl.graphbolt.ItemSet(torch.arange(10))
+    minibatch_dp = dgl.graphbolt.MinibatchSampler(itemset, batch_size=2)
+    fetcher_dp = dgl.graphbolt.FeatureFetcher(minibatch_dp, fetch_func)
```

Comment:
For fetcher, do we need extra flexibility such that user can define fetch_func? Can we directly passing in the feature store?

Reply:
This will be hard as we need to define the protocol on how to return the features and how to select the appropriate keys to read/write.